js_modules/dagster-ui/packages/ui-components/src/components/createToaster.tsx

```diff
@@ -1,15 +1,32 @@
 // eslint-disable-next-line no-restricted-imports
 import {IToasterProps, Toaster} from '@blueprintjs/core';
 import * as React from 'react';
-import {createRoot} from 'react-dom/client';
+
+type PortalProvider = (node: React.ReactNode, container: HTMLElement, key?: string) => void;
+const queue: Parameters<PortalProvider>[] = [];
+let _portalProvider: PortalProvider = (node, container, key) => {
+  queue.push([node, container, key]);
+};
+
+let _baseHref = '';
+
+export const registerPortalProvider = (portalProvider: PortalProvider, baseHref: string) => {
```

Comment:
I can probably remove this queue, but this is just in case `registerPortalProvider` gets called after a component calls `createToaster`. This could happen if a module's factory executes before AppProvider's `useLayoutEffect` fires. I think it's plausible it could happen since I saw createToaster being called in module factories rather than inside of render callbacks